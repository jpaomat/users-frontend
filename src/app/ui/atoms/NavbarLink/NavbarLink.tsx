import { NavLink } from 'react-router-dom';
import { NavbarLinkProps } from '../../../interfaces';

export const NavbarLink = ({ label, redirectTo }: NavbarLinkProps) => {
  
	return (
		<NavLink
			className={({ isActive }) => `nav-link ${isActive ? 'active' : ''}`}
			to={redirectTo}
		>
			{label}
		</NavLink>
	);
};
