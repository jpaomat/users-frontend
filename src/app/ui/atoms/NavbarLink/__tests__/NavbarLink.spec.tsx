import { fireEvent, render, screen } from '@testing-library/react';
import { NavbarLink } from '../NavbarLink';
import { MemoryRouter } from 'react-router-dom';

describe('<NavbarLink /> component', () => {

	beforeEach(()=> jest.clearAllMocks);
	
	test('should render component', () => {
		const { container } = render(
			<MemoryRouter>
				<NavbarLink text='test' path='/test'/>
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});

	test('should show the text linl', () => {
		const dataLink = {
			text: 'Productos',
			path:'/products'
		};
		render(
			<MemoryRouter>
				<NavbarLink {...dataLink} />
			</MemoryRouter>
		);
		fireEvent.click(screen.getByText(dataLink.text));
		expect(screen.getByText(dataLink.text)).toBeTruthy();
	});
});
