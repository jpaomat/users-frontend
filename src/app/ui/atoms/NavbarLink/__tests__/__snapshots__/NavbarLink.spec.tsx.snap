// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarLink /> component should render component 1`] = `
<div>
  <a
    class="nav-link "
    href="/test"
  >
    test
  </a>
</div>
`;
